survival4d/nn/torch/__init__.py

```diff
@@ -41,7 +44,14 @@ def train_nn(xtr, ytr, batch_size, n_epochs, model_name, lr_exp, alpha, weight_d
     model = model_factory(model_name, **model_kwargs)
 
     dataset = TensorDataset(torch.from_numpy(X_tr).cuda(), torch.from_numpy(E_tr).cuda(), torch.from_numpy(TM_tr).cuda())
-    dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=True)
+
+    class CustomBatchSampler(BatchSampler):
+        def __iter__(self):
+            for res in super().__iter__():
+                yield sorted(res)
```

Comment:
sorting according to? I think this sort bases on the tuple of x,e,t, it should sort based on t